### ISSUE TITLE
Incorrect spelling suggestion in French

### ISSUE THREAD

**Issue by dpelle** [languagetool-org/languagetool#912]
I see that spelling corrections in French does not propose the correct
suggestion in this sentence, as was mentioned by @mattpepin earlier in https://github.com/languagetool-org/languagetool/issues/850 (but that issue was closed):
```
$ echo "Ca fonctionne." | java -jar /home/pel/sb/languagetool/languagetool-standalone/target/LanguageTool-4.1-SNAPSHOT/LanguageTool-4.1-SNAPSHOT/languagetool-commandline.jar -c utf-8 -l fr -
Expected text language: French
Working on STDIN...
1.) Line 1, column 1, Rule ID: FR_SPELLING_RULE
Message: Faute de frappe possible trouvée
Suggestion: CA; C a
Ca fonctionne. 
^^             
Time: 231ms for 1 sentences (4.3 sentences/sec)
```

LT suggests "CA" and "C a" which makes no sense (they are not French words).
Well, "CA" is the French dicollecte dictionary, but it's only an obscure abbreviation.

LT should instead suggest "Ça".  The only difference with the erroneous word "Ca"
is a diacritics (cedilla).  Diacritics differences should have a low penalty for
suggestions, yet "Ça" does not show up as suggestion.

If I type the sentence without mistake as below, LT does not show errors as
expected, so it knows that "Ça" is a correct spelling:
```
$ echo "Ça fonctionne." | java -jar /home/pel/sb/languagetool/languagetool-standalone/target/LanguageTool-4.1-SNAPSHOT/LanguageTool-4.1-SNAPSHOT/languagetool-commandline.jar -c utf-8 -l fr -
Expected text language: French
Working on STDIN...
Time: 222ms for 1 sentences (4.5 sentences/sec)
```

I tried another word with a cedilla: *décu* instead of the correct "déçu" (= disappointed):

```
$ echo "Décu" | java -jar /home/pel/sb/languagetool/languagetool-standalone/target/LanguageTool-4.1-SNAPSHOT/LanguageTool-4.1-SNAPSHOT/languagetool-commandline.jar -c utf-8 -l fr -
Expected text language: French
Working on STDIN...
1.) Line 1, column 1, Rule ID: FR_SPELLING_RULE
Message: Faute de frappe possible trouvée
Suggestion: Vécu; Écu; D'écu; Sécu; Élu; Baisse; Laisse; Laissé; Caisse; Laissa; Vaincu; Dick; Reich; Baissé; Déchu; Déçu; Déco; L'écu; Écus; Daisy
Décu 
^^^^ 
```

There I see that "Déçu" is suggested, but only is position 16 
which is bad. The difference between ç and c should be small because
it's only a diacritic mistake. So the suggestion "Déçu" should come
at the top in my opinion.

Besides, most of the corrections it proposes are way too far
both in term of Levenshtein distance or in term of pronunciation
to the erroneous words *decu*. I mean, seeing the word
"Reich" being proposed as suggestion for "Décu" is rather odd,
among others.

Perhaps that can be tuned but I am not  familiar with the way
spelling suggestion works.

**Comment by dpelle** [comment]
I just found this comment in file languagetool-language-modules/fr/src/main/resources/org/languagetool/resource/fr/hunspell/fr_FR.info

```
# ignore-diacritics=false speeds up building the suggestions by a factor of about 2:
fsa.dict.speller.ignore-diacritics=false
```

I suspect that this may be causing the issue.  I have not tried
replace the 'false' with 'true' there to confirm, as I don't know yet how
to rebuild the dictionary.

As I wrote in the ticket description, diacritics should have only a small
penalty when ranking spelling corrections. I don't think we can afford
fsa.dict.speller.ignore-diacritics=false  if it causes incorrect suggestions.
I'd rather live with slower suggestions. I assume it will still be faster
than Hunspell suggestions.  I also assume that the cedilla will then be
treated as a diacritic.

**Comment by danielnaber** [comment]
I just check it, this issue can be fixed by using morfologik-stemming 2.1.5-SNAPSHOT **and** `fsa.dict.speller.ignore-diacritics=true`. So first, we need a new Morfologik release, I'll ask Dawid Weiss if that's possible.

**Comment by danielnaber** [comment]
I've upgraded to Morfologik 2.1.5 - @jaumeortola a change was needed to make Catalan test work again, it might be a small regression: https://github.com/languagetool-org/languagetool/commit/7847c9e3d9bfac3158efca23474eaf1f1d047c5a#diff-aef53c331bef8fc633052c18907a33d1L231
